### PR TITLE
Add wireguard role to partition

### PR DESCRIPTION
**Pull request by mwindower**

### CODE REVIEW

**Comment by majst01** [comment]
Unsure if we should add this here, i would like to prefer the tailscale approach. But does not hurt.

**Review by majst01 — APPROVED** [review]

File: partition/roles/wireguard/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+- name: check mandatory variables for this role are set
+  assert:
+    fail_msg: "not all mandatory variables given, check role documentation"
+    quiet: yes
+    that:
+      - wireguard_ip is not none
+      - wireguard_clients is not none
+      - wireguard_clients|length > 0
```
Comment (Gerrit91) [inline partition/roles/wireguard/tasks/main.yaml:9]: Just `wireguard_clients` is enough, it does exactly this

File: partition/roles/wireguard/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+- name: install wireguard
+  apt:
+    name: wireguard
+    update_cache: yes
+    state: latest
+  with_items:
```
Comment (Gerrit91) [inline partition/roles/wireguard/tasks/main.yaml:16]: Deprecated for `loop`

File: partition/roles/wireguard/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    - wireguard
+    - wireguard-dkms
+    - wireguard-tools
```
Comment (Gerrit91) [inline partition/roles/wireguard/tasks/main.yaml:19]: `item` is not used, this loop does three times the same?

File: partition/roles/wireguard/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+  - name: set public key fact
+    set_fact:
+      wg_public_key: "{{ wg_public_key_result.stdout }}"
+
+  when: not private_key_file_stat.stat.exists
+  tags: wg-generate-keys
+
+- name: write WireGuard configuration
+  block:
```
Comment (Gerrit91) [inline partition/roles/wireguard/tasks/main.yaml:75]: `block` is used but no reason to do so

File: partition/roles/wireguard/tasks/main.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+- name: install wireguard
+  apt:
+    name: wireguard
+    update_cache: yes
+    state: latest
+  with_items:
+    - wireguard
+    - wireguard-dkms
+    - wireguard-tools
+  tags: wg-install
```
Comment (Gerrit91) [inline partition/roles/wireguard/tasks/main.yaml:20]: Do we need the tags on these tasks? Somehow we do not have any tags anywhere else in the roles of this repo.

**Pull request merged**